app/src/main/java/com/kkolontay/theworld/ui/screens/WorldNavigation.kt

```diff
@@ -27,70 +30,60 @@ enum class AppScreens() {
 
 @Composable
 fun WorldNavigation(
-   // viewModel: OrderViewModel = viewModel(),
+    viewModel: CountryViewModel = androidx.lifecycle.viewmodel.compose.viewModel(),
     navController: NavHostController = rememberNavController(),
     context: Context
 ) {
-    val backStackEntry by navController.currentBackStackEntryAsState()
-    val currentScreen = AppScreens.valueOf(
-        backStackEntry?.destination?.route ?: AppScreens.ListCountry.name
-    )
+
     var title = rememberSaveable {
-        mutableStateOf("List")
+        mutableStateOf("")
+    }
+    var choosenCountry = remember {
+        mutableStateOf<Country>(Country(name = CountryName(common = "some"), capital = listOf("other"), population = 34, area = 45.0, flags = CountryFlags(png = "some")))
     }
```

Comment:
You shouldn't need `title` as a remembered variable here in the navigation layer.  Instead, you could create scaffolds with `AppBar`s in each of the respective screen routes below.  The alternative would be to keep `title` but then clear it when navigating up and back from the details screen, otherwise the country name remains on the list screen.  Doing the latter though gives almost too much responsibility to this navigation layer.  Really all that should be happening here is setting up routes and navigation, nothing else.

Reply:
You could also remove the need for a remembered `chosenCountry`. Instead pass in an argument of the country name:

```kotlin
composable(
    "${AppScreens.CountryDetail.name}/{countryName}",
    arguments = listOf(navArgument("countryName") { type = NavType.StringType }),
) {
    // The !! is fine here as we expect this to always be set based on the route name.
    val countryName = backStackEntry.arguments!!.getString("countryName")
    
    CountryItemDetail(country = countryName, taps = tap.value, back = back.value, refresh = {
        ...
```

Then have a viewmodel for your details screen that reaches out to a CountryRepository where it can fetch the specific country based on the name. This thread overall is more minor at the moment, as we are only just talking about viewmodel and repository this week.  No changes are required here, just suggestions.